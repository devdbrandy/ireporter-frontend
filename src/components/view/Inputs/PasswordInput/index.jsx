/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/interactive-supports-focus */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import PropType from 'prop-types';

const PasswordInput = ({
  name,
  placeholder,
  handleOnChange,
  label,
  labelText,
  pattern,
  title,
  minLength
}) => {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <label htmlFor={label}>
      {labelText}
      <div className="input-wrapper">
        <input
          type={showPassword ? 'text' : 'password'}
          name={name}
          placeholder={placeholder}
          style={{ borderRight: 0 }}
          onChange={event => handleOnChange(event)}
          required
          pattern={pattern}
          title={title}
          minLength={minLength}
        />
        <i
          className={`fas ${showPassword ? 'fa-eye' : 'fa-eye-slash'} icon`}
          onClick={() => setShowPassword(!showPassword)}
        />
      </div>
    </label>
  );
};

PasswordInput.propTypes = {
  name: PropType.string.isRequired,
  placeholder: PropType.string.isRequired,
  handleOnChange: PropType.func.isRequired,
  label: PropType.string,
  labelText: PropType.string,
  pattern: PropType.string,
  title: PropType.string,
  minLength: PropType.string,
};

PasswordInput.defaultProps = {
  label: '',
  labelText: '',
  pattern: null,
  title: null,
  minLength: null,
};

export default PasswordInput;
