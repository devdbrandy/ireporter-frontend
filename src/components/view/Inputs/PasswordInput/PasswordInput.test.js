import React from 'react';
import { shallow } from 'enzyme';
import PasswordInput from './index';

describe('<PasswordInput />', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      name: '',
      placeholder: '',
      handleOnChange: jest.fn()
    };
    const enzymeWrapper = shallow(<PasswordInput {...props} />);
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
