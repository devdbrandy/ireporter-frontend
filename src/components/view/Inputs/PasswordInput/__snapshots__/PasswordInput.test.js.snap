// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> renders without crashing given the required props 1`] = `
<label
  htmlFor=""
>
  <div
    className="input-wrapper"
  >
    <input
      minLength={null}
      name=""
      onChange={[Function]}
      pattern={null}
      placeholder=""
      required={true}
      style={
        Object {
          "borderRight": 0,
        }
      }
      title={null}
      type="password"
    />
    <i
      className="fas fa-eye-slash icon"
      onClick={[Function]}
    />
  </div>
</label>
`;
